### DISCUSSION TITLE
Use of Registered IO on Windows?

### DISCUSSION THREAD

**Discussion by DrJWCain** [microsoft/msquic#1675] [Ideas]
Hi,

Very excited about RFC 9000 et al. Also reviewing this code makes great reading - thanks for sharing!
I have a (potentially naïve) question: would the IOCP use on Windows benefit from using Registered IO? 
I can see the IOCP calls are made per processor (with thread affinity), to get RSS working efficiently.
I'm just wondering if the code would benefit from RIO as well? 

Cheers, 

James.

**Comment by nibanks** [comment]
Hi @DrJWCain, RIO does have some benefits but the major downside to its usage is that you must preallocate all the receive buffer memory.  We prefer not to have to do this, as we don't know how much we'd have to preallocate. As you can see from [our dashboard](https://microsoft.github.io/msquic/index.html) perf is really quite good with the current model.

**Reply by DrJWCain** [reply]
I understand - Thanks for the quic (sorry) reply :-)
